base/strings/basic.jl

```diff
@@ -140,6 +140,32 @@ prevind(s::AbstractArray   , i::Integer) = i-1
 nextind(s::DirectIndexString, i::Integer) = i+1
 nextind(s::AbstractArray   , i::Integer) = i+1
 
+function prevind(s::String, i::Integer)
+    data = s.data
+    stop = endof(s.data)
+    if i > stop
+        return endof(s)
+    end
+    i -= oftype(i, 1)
```

Comment:
At least the functions should be made type stable. Currently if a bigger type than machine `Int` is passed in, the return type is a `Union` and that's not good. I'll make both functions (including the generic ones) return `Int` always.